stdlib/_ast.pyi

```diff
@@ -59,27 +60,39 @@ class Expression(mod):
 class stmt(AST): ...
 
 class FunctionDef(stmt):
-    if sys.version_info >= (3, 10):
+    if sys.version_info >= (3, 12):
+        __match_args__ = ("name", "type_params", "args", "body", "decorator_list", "returns", "type_comment")
```

Comment:
I didn't realise you were adding type_params to the__match_args__ for various AST classes. I guess that makes sense, but it feels like it could be quite a breaking change if anybody had been matching FuncDef or ClassDef nodes with "positional arguments" in their `case` statements, e.g.

```python
match node:
    case FunctionDef("foo", ast.arguments(), []):  # <- would have matched on 3.11, not on 3.12!
        ...
```

Feels like it would be much less breaking if type_params was added to the end of `__match_args__` rather than inserted into the middle?

Reply:
At the very least, seems like this change to `__match_args__` should be clearly documented in https://github.com/python/cpython/pull/104642

Reply:
Sure, I'll open an issue

Reply:
I filed:

- https://github.com/python/cpython/issues/104799